### DISCUSSION TITLE
How can i use the SubjectContains method programmatically ?

### DISCUSSION THREAD

**Discussion by ssousa97** [jstedfast/MailKit#1834] [Q&A] [answered]
I have a list of strings that contains keywords i'm interested on filtering, something like:
`var _appsToInclude = ["APP1", "ABC2", "HYX2"...]`

The emails i want have the subject like "Email APP1". What i want is to filter all the emails that contains any of the words in `appsToInclude`. The problem is the only to do it (as far as i know) is manually write like this:

```
var query = SearchQuery.SubjectContains("Email")
.And(
   SearchQuery.SubjectContains("APP1")
  .Or(SearchQuery.SubjectContains("ABC2"))
  .Or(SearchQuery.SubjectContains("HYX2"))
...
```

Do we have something like `SearchQuery.SubjectContains("APP A1|A2|A3...")` or any other way to fill this based on the initial array ? 
I know i can fetch all emails with only "Email" on the subject and filter when get the response, but that's not ideal for me

**Comment by jstedfast** [comment]
```csharp
SearchQuery query = SearchQuery.SubjectContains(appsToInclude[0]);
for (int i = 1; i < appsToInclude.Count; i++)
    query = query.Or (SearchQuery.SubjectContains (appsToInclude[i]));

query = SearchQuery.SubjectContains ("Email").And (query);
```

**Reply by ssousa97** [reply]
I tried using it, but 
`Cannot convert source type 'MailKit.Search.BinarySearchQuery' to target type 'MailKit.Search.TextSearchQuery'`

**Reply by ssousa97** [reply]
had to do it like this:
```
var query = SearchQuery.SubjectContains(_config.AppsToInclude[0]).Or(SearchQuery.SubjectContains(_config.AppsToInclude[2]));
        for (var i = 2; i < _config.AppsToInclude.Length; i++) {
            query = query.Or(SearchQuery.SubjectContains(_config.AppsToInclude[i]));
        }
        query = SearchQuery.SubjectContains("Email").And(query);
```

**Reply by jstedfast** [reply, marked as answer]
Instead of `var query = ...` in my original solution, use `SearchQuery query = ...`

**Reply by ssousa97** [reply]
Wow, it was that simple haha
Thank you !
